x/crosschain/keeper/keeper_tss_voter.go

```diff
@@ -89,13 +89,15 @@ func (k msgServer) CreateTSSVoter(goCtx context.Context, msg *types.MsgCreateTSS
 	// Set TSS only on success, set Keygen either way.
 	// Keygen block can be updated using a policy transaction if keygen fails
 	if ballot.BallotStatus != observerTypes.BallotStatus_BallotFinalized_FailureObservation {
-		k.SetTSS(ctx, types.TSS{
+		tss := types.TSS{
 			TssPubkey:           msg.TssPubkey,
 			TssParticipantList:  keygen.GetGranteePubkeys(),
 			OperatorAddressList: ballot.VoterList,
 			FinalizedZetaHeight: ctx.BlockHeight(),
 			KeyGenZetaHeight:    msg.KeyGenZetaHeight,
-		})
+		}
+		k.SetTSS(ctx, tss)
+		k.SetTSSHistory(ctx, tss)
```

Comment:
Maybe we could implement a method `AppendTSS` that would make the atomic operation of adding a new TSS.